AppleClang fails to compile tsq.hpp with undeclared identifier 'TF_DEFAULT_UNBOUNDED_TASK_QUEUE_LOG_SIZE' and `TF_DEFAULT_UNBOUNDED_TASK_QUEUE_LOG_SIZE`

**Describe the bug**
Failed to build TaskFlow v3.8.0 with AppleClang 16.0.0

**To Reproduce**

Build C++ source file with `#include <taskflow/algorithm/for_each.hpp>` using Xcode and TaskFlow with options:
- TF_BUILD_BENCHMARKS=OFF
- TF_BUILD_CUDA=OFF
- TF_BUILD_SYCL=OFF
- TF_BUILD_TESTS=OFF
- TF_BUILD_EXAMPLES=OFF

```cpp
/Users/egorodet/git/MethanePowered/MethaneKit/Build/Output/CLion/macOS/MTL/Release/Modules/Graphics/RHI/Base/CMakeFiles/MethaneGraphicsRhiBase.dir/Unity/unity_5_cxx.cxx
In file included from /Users/egorodet/git/MethanePowered/MethaneKit/Build/Output/CLion/macOS/MTL/Release/Modules/Graphics/RHI/Base/CMakeFiles/MethaneGraphicsRhiBase.dir/Unity/unity_5_cxx.cxx:7:
In file included from /Users/egorodet/git/MethanePowered/MethaneKit/Modules/Graphics/RHI/Base/Sources/Methane/Graphics/Base/DescriptorManager.cpp:30:
In file included from /Users/egorodet/git/MethanePowered/MethaneKit/Build/Output/ExternalsCache/taskflow/572b99e63ff4a9f8bc79c3e0631c315b8b9d9b42/taskflow/algorithm/for_each.hpp:3:
In file included from /Users/egorodet/git/MethanePowered/MethaneKit/Build/Output/ExternalsCache/taskflow/572b99e63ff4a9f8bc79c3e0631c315b8b9d9b42/taskflow/algorithm/launch.hpp:4:
In file included from /Users/egorodet/git/MethanePowered/MethaneKit/Build/Output/ExternalsCache/taskflow/572b99e63ff4a9f8bc79c3e0631c315b8b9d9b42/taskflow/algorithm/../core/async.hpp:3:
In file included from /Users/egorodet/git/MethanePowered/MethaneKit/Build/Output/ExternalsCache/taskflow/572b99e63ff4a9f8bc79c3e0631c315b8b9d9b42/taskflow/algorithm/../core/executor.hpp:3:
In file included from /Users/egorodet/git/MethanePowered/MethaneKit/Build/Output/ExternalsCache/taskflow/572b99e63ff4a9f8bc79c3e0631c315b8b9d9b42/taskflow/algorithm/../core/observer.hpp:3:
In file included from /Users/egorodet/git/MethanePowered/MethaneKit/Build/Output/ExternalsCache/taskflow/572b99e63ff4a9f8bc79c3e0631c315b8b9d9b42/taskflow/algorithm/../core/task.hpp:3:
In file included from /Users/egorodet/git/MethanePowered/MethaneKit/Build/Output/ExternalsCache/taskflow/572b99e63ff4a9f8bc79c3e0631c315b8b9d9b42/taskflow/algorithm/../core/graph.hpp:22:
/Users/egorodet/git/MethanePowered/MethaneKit/Build/Output/ExternalsCache/taskflow/572b99e63ff4a9f8bc79c3e0631c315b8b9d9b42/taskflow/algorithm/../core/tsq.hpp:90:49: error: use of undeclared identifier 'TF_DEFAULT_UNBOUNDED_TASK_QUEUE_LOG_SIZE'
   90 |   explicit UnboundedTaskQueue(int64_t LogSize = TF_DEFAULT_UNBOUNDED_TASK_QUEUE_LOG_SIZE);
      |                                                 ^
/Users/egorodet/git/MethanePowered/MethaneKit/Build/Output/ExternalsCache/taskflow/572b99e63ff4a9f8bc79c3e0631c315b8b9d9b42/taskflow/algorithm/../core/tsq.hpp:288:40: error: use of undeclared identifier 'TF_DEFAULT_BOUNDED_TASK_QUEUE_LOG_SIZE'
  288 | template <typename T, size_t LogSize = TF_DEFAULT_BOUNDED_TASK_QUEUE_LOG_SIZE>
      |                                        ^
In file included from /Users/egorodet/git/MethanePowered/MethaneKit/Build/Output/CLion/macOS/MTL/Release/Modules/Graphics/RHI/Base/CMakeFiles/MethaneGraphicsRhiBase.dir/Unity/unity_5_cxx.cxx:7:
In file included from /Users/egorodet/git/MethanePowered/MethaneKit/Modules/Graphics/RHI/Base/Sources/Methane/Graphics/Base/DescriptorManager.cpp:30:
In file included from /Users/egorodet/git/MethanePowered/MethaneKit/Build/Output/ExternalsCache/taskflow/572b99e63ff4a9f8bc79c3e0631c315b8b9d9b42/taskflow/algorithm/for_each.hpp:3:
In file included from /Users/egorodet/git/MethanePowered/MethaneKit/Build/Output/ExternalsCache/taskflow/572b99e63ff4a9f8bc79c3e0631c315b8b9d9b42/taskflow/algorithm/launch.hpp:4:
In file included from /Users/egorodet/git/MethanePowered/MethaneKit/Build/Output/ExternalsCache/taskflow/572b99e63ff4a9f8bc79c3e0631c315b8b9d9b42/taskflow/algorithm/../core/async.hpp:3:
In file included from /Users/egorodet/git/MethanePowered/MethaneKit/Build/Output/ExternalsCache/taskflow/572b99e63ff4a9f8bc79c3e0631c315b8b9d9b42/taskflow/algorithm/../core/executor.hpp:3:
In file included from /Users/egorodet/git/MethanePowered/MethaneKit/Build/Output/ExternalsCache/taskflow/572b99e63ff4a9f8bc79c3e0631c315b8b9d9b42/taskflow/algorithm/../core/observer.hpp:4:
/Users/egorodet/git/MethanePowered/MethaneKit/Build/Output/ExternalsCache/taskflow/572b99e63ff4a9f8bc79c3e0631c315b8b9d9b42/taskflow/algorithm/../core/worker.hpp:76:5: error: too few template arguments for class template 'BoundedTaskQueue'
   76 |     BoundedTaskQueue<Node*> _wsq;
      |     ^
/Users/egorodet/git/MethanePowered/MethaneKit/Build/Output/ExternalsCache/taskflow/572b99e63ff4a9f8bc79c3e0631c315b8b9d9b42/taskflow/algorithm/../core/tsq.hpp:289:7: note: template is declared here
  288 | template <typename T, size_t LogSize = TF_DEFAULT_BOUNDED_TASK_QUEUE_LOG_SIZE>
      | ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  289 | class BoundedTaskQueue {
      |       ^
3 errors generated.
```

**Build Environment**
- Apple clang version 16.0.0 (clang-1600.0.26.6)
- Target: arm64-apple-darwin24.2.0

--- a/taskflow/core/tsq.hpp
+++ b/taskflow/core/tsq.hpp
@@ -1,5 +1,6 @@
 #pragma once
 
+#include "../taskflow.hpp"
 #include "../utility/macros.hpp"
 #include "../utility/traits.hpp"
 

--- a/taskflow/core/tsq.hpp
+++ b/taskflow/core/tsq.hpp
@@ -1,6 +1,5 @@
 #pragma once
 
-#include "../taskflow.hpp"
 #include "../utility/macros.hpp"
 #include "../utility/traits.hpp"
 
@@ -8,6 +7,26 @@
 @file tsq.hpp
 @brief task queue include file
 */
+
+#ifndef TF_DEFAULT_BOUNDED_TASK_QUEUE_LOG_SIZE 
+/**
+@def TF_DEFAULT_BOUNDED_TASK_QUEUE_LOG_SIZE
+
+This macro defines the default size of the bounded task queue in Log2. 
+Bounded task queue is used by each worker.
+*/
+  #define TF_DEFAULT_BOUNDED_TASK_QUEUE_LOG_SIZE 8
+#endif
+
+#ifndef TF_DEFAULT_UNBOUNDED_TASK_QUEUE_LOG_SIZE 
+/**
+@def TF_DEFAULT_UNBOUNDED_TASK_QUEUE_LOG_SIZE
+
+This macro defines the default size of the unbounded task queue in Log2.
+Unbounded task queue is used by the executor.
+*/
+  #define TF_DEFAULT_UNBOUNDED_TASK_QUEUE_LOG_SIZE 10
+#endif
 
 namespace tf {
 

--- a/taskflow/taskflow.hpp
+++ b/taskflow/taskflow.hpp
@@ -7,27 +7,6 @@
 // + TF_ENABLE_ATOMIC_NOTIFIER : enable atomic notifier (required C++20)
 //
 
-#ifndef TF_DEFAULT_BOUNDED_TASK_QUEUE_LOG_SIZE 
-/**
-@def TF_DEFAULT_BOUNDED_TASK_QUEUE_LOG_SIZE
-
-This macro defines the default size of the bounded task queue in Log2. 
-Bounded task queue is used by each worker.
-*/
-  #define TF_DEFAULT_BOUNDED_TASK_QUEUE_LOG_SIZE 8
-#endif
-
-#ifndef TF_DEFAULT_UNBOUNDED_TASK_QUEUE_LOG_SIZE 
-/**
-@def TF_DEFAULT_UNBOUNDED_TASK_QUEUE_LOG_SIZE
-
-This macro defines the default size of the unbounded task queue in Log2.
-Unbounded task queue is used by the executor.
-*/
-  #define TF_DEFAULT_UNBOUNDED_TASK_QUEUE_LOG_SIZE 10
-#endif
-
-
 #include "core/executor.hpp"
 #include "core/async.hpp"
 

